### DISCUSSION TITLE
Connessione al MSSQL docker

### DISCUSSION THREAD

**Discussion by Simpz78** [alkampfergit/personal-blog#2] [Q&A]
Ciao,
ho letto il tuo articolo sulla connessione al Docker di MSSQL sotto ubuntu per il .NET ma non riesco a connettermi...
l'errore che mi viene fuori se provo a connettermi con la connection string dentro al mio progetto .NET è sempre questo:

Error Message:
   System.Data.SqlClient.SqlException : A connection was successfully established with the server, but then an error occurred during the pre-login handshake. (provider: TCP Provider, error: 0 - Success)

viene sempre sul connection.Open()

la connection string è: Server=localhost,5000;Database=Master;User ID=SA;Password=sqlPw3$secure;Integrated Security=False
ma anche se tolgo l'integration security è lo stesso.

Hai qualche suggerimento? Il progetto è in .NET 5.0, la macchina container è ubuntu-latest e l'mssql è il 2017-latest-unbuntu

Grazie!

il progetto è qui https://github.com/Simpz78/PingDockerTest

Ciao!

**Comment by alkampfergit** [comment]
Probabilmente dovresti usare il comando

Socket nomecontainer logs

Per vedere i log del container SQL, probabilmente ha avuto qualche problema in avvio.

**Reply by Simpz78** [reply]
Grazie per la dritta. Il problema è che non davo il tempo fisico al server SQL di prepararsi. Ho risolto mettendo un time nella action di 30 secondi.
Adesso ho un problema diverso che in solcale non ho. Sto cercando di caricare nel docker mssql un file .sql di script per "preparare" il db ai test.
La copia direi che risce, con questo comando qui:
docker cp $GITHUB_WORKSPACE/test.sql mssql:/tmp/test.sql

ma il comando per farlo partire mi dà sempre l'errore di TTY
docker exec -it msssql /opt/mssql-tools/bin/sqlcmd -i /tmp/test.sql -U SA -P 'sqlPw3$secure'
"The input device is not a TTY".

in locale (uso powershell su VS Code) non dà questo errore

Ho provato a mettere --tty come opzione a docker exec ma nulla, e anche a mettere /bin/bash così
docker exec -it msssql /bin/bash /opt/mssql-tools/bin/sqlcmd -i /tmp/test.sql -U SA -P 'sqlPw3$secure'
ma l'errore è sempre quello.
Suggetimenti?

Grazie!

**Reply by alkampfergit** [reply]
Non sono un guru di docker, ma cosi ad occhio il parametro -it te lo fa girare interattivo il comando e secondo me potrebbe essere la causa dell'errore, visto che il messaggio lamenta un errore di console. Toglierei il parametro -it

**Reply by Simpz78** [reply]
Infatti sostituendo -i al -it è tutto ok. Quindi il comando sarebbe:
docker exec -i msssql /bin/bash /opt/mssql-tools/bin/sqlcmd -i /tmp/test.sql -U <MyUser> -P '<MYPASS>'
